src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,469 @@
+        /// <summary>
+        /// CPUS-PORT base address.
+        /// </summary>
+        protected virtual int CpusPortBaseAddess { get; }
+        // final_address = mapped_address + (target_address & map_mask) https://stackoverflow.com/a/37922968
+        private readonly int _mapMask = Environment.SystemPageSize - 1;
+
+        private static readonly object s_initializationLock = new object();
+        private static readonly object s_sysFsInitializationLock = new object();
+        private const string GpioMemoryFilePath = "/dev/mem";
+        private UnixDriver _interruptDriver;
+        private readonly IDictionary<int, PinState> _pinModes = new Dictionary<int, PinState>();
+
+        protected SunxiDriver() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SunxiDriver"/>.
+        /// </summary>
+        /// <param name="cpuxPortBaseAddess">CPUX-PORT base address (This can be find in the corresponding SoC datasheet).</param>
+        /// <param name="cpusPortBaseAddess">CPUS-PORT base address (This can be find in the corresponding SoC datasheet).</param>
+        public SunxiDriver(int cpuxPortBaseAddess, int cpusPortBaseAddess)
+        {
+            CpuxPortBaseAddess = cpuxPortBaseAddess;
+            CpusPortBaseAddess = cpusPortBaseAddess;
+        }
+
+        /// <summary>
+        /// The number of pins provided by the driver.
+        /// </summary>
+        protected internal override int PinCount => throw new PlatformNotSupportedException("This driver is generic so it can not enumerate how many pins are available.");
+
+        /// <summary>
+        /// Converts a board pin number to the driver's logical numbering scheme.
+        /// </summary>
+        /// <param name="pinNumber">The board pin number to convert.</param>
+        /// <returns>The pin number in the driver's logical numbering scheme.</returns>
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        /// <summary>
+        /// Opens a pin in order for it to be ready to use.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        protected internal override void OpenPin(int pinNumber)
+        {
+            Initialize();
+            SetPinMode(pinNumber, PinMode.Input);
+        }
+
+        /// <summary>
+        /// Closes an open pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        protected internal override void ClosePin(int pinNumber)
+        {
+            if (_pinModes.ContainsKey(pinNumber))
+            {
+                if (_pinModes[pinNumber].InUseByInterruptDriver)
+                {
+                    _interruptDriver.ClosePin(pinNumber);
```

Comment:
I wonder if we should make the SunxiDriver derive from UnixDriver instead that way it will be able to call these methods.

Reply:
`UnixDriver` is an abstract class🤔

Reply:
If I use `ProjectReference`, how to solve `CS1540` error😅 @joperezr